master/internal/rm/agent_resource_manager.go

```diff
@@ -95,16 +96,38 @@ func (a AgentResourceManager) IsReattachEnabledForRP(ctx actor.Messenger, rpName
 	return false
 }
 
+// CheckMaxSlotsExceeded checks if the job exceeded the maximum number of slots.
+func (a AgentResourceManager) CheckMaxSlotsExceeded(
+	ctx actor.Messenger, name string, slots int,
+) (bool, error) {
+	agents, err := a.GetAgents(ctx, &apiv1.GetAgentsRequest{})
+	if err != nil {
+		return false, fmt.Errorf("unable to query agents for resource pool %v", name)
+	}
+	maxSlots := 0
+	for _, agent := range agents.Agents {
+		for _, resPool := range agent.ResourcePools {
+			if resPool == name {
+				maxSlots += len(agent.Slots)
+			}
+		}
+	}
+	if slots > maxSlots {
+		return true, nil
+	}
+	return false, nil
+}
+
 // ResolveResourcePool fully resolves the resource pool name.
 func (a AgentResourceManager) ResolveResourcePool(
-	ctx actor.Messenger, name string, slots int, command bool,
+	ctx actor.Messenger, name string, slots int,
 ) (string, error) {
 	// If the resource pool isn't set, fill in the default at creation time.
 	if name == "" && slots == 0 {
 		req := sproto.GetDefaultAuxResourcePoolRequest{}
 		resp, err := a.GetDefaultAuxResourcePool(ctx, req)
 		if err != nil {
-			return "", fmt.Errorf("defaulting to aux pool: %w", err)
+			return resp.PoolName, fmt.Errorf("defaulting to aux pool: %w", err)
```

Comment:
why this change? without looking, i would i think in the event of an an error, checking `resp.PoolName` may be a nil pointer panic.

Reply:
the reason i'm pushing returning explicit nil or "" - not returning anything meaningful when you error - is that functions/apis that allow partial success are easy to misuse.